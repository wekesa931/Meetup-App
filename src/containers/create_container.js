import React, { Component } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { createMeetupItem,meetupListAll } from '../actions';

class CreateContainer extends Component {
    createMeetupItem = async (mtp,props) => {
        await props.createMeetupItem(mtp)
        props.meetupListAll()
    }
    render() { 
        return (
            <div>
                <button type='button'
                    style={{
                        background:'#aef1ae',
                        padding:'5px',
                        margin: '5px'
                    }}
                    onClick={()=>this.createMeetupItem(this.props.mtp,this.props)}
                >Create Meetup</button>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        meetup: state.meetupList.meetupList
    }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
        createMeetupItem,meetupListAll
    },dispatch)
}
 
export default connect(mapStateToProps,mapDispatchToProps)(CreateContainer);