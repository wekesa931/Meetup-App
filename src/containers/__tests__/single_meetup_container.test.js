import React from "react";
import { Provider } from "react-redux";
import Enzyme, { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import {SingleMeetup} from '../create_container';
import EnzymeAdapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new EnzymeAdapter() });

const mockStore = configureMockStore();
const store = mockStore({});

describe('Get single Meetup container', () => {
    // const wrapper = shallow(<Provider store={store}>
    //                             <SingleMeetup />
    //                         </Provider>);
    // it('renders correctly', () => {
    //     expect(wrapper).toMatchSnapshot()
    // })

    let props;
    let wrapper;
    let spy;
    beforeEach(() => {
        props = {
            meetups: [
                {
                    id: 1,
                    title: "Another test meetup"
                }
            ]
        };
        props.singleMeetupItem = jest.fn();
        wrapper = shallow(
            <Provider store={store}>
                <SingleMeetup {...props}/> 
            </Provider> );
        spy = jest.spyOn(wrapper.instance().props.children.props, 'singleMeetupItem');
    });
    
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it("fetches meetups on load", () => {
        spy()
        expect(spy).toHaveBeenCalled();
    });
    
})