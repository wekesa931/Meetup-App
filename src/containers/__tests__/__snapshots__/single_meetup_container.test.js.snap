// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get single Meetup container renders correctly 1`] = `ShallowWrapper {}`;
